### PR TITLE
web: Support salign, quality, & scale embed/object attributes (part of #4258)

### PR DESCRIPTION
**Pull request by danielhjacobs**
Good test for salign (recently mentioned in Discord):

https://deconcept.com/

If anyone knows a good test for scale, let me know. I tested it locally and it seems to work. According to #4601, quality does nothing currently in Ruffle, but I still added support for it since implementing all three of these attributes followed the exact same process so it made sense to do so.

To be added to [the Wiki](https://github.com/ruffle-rs/ruffle/wiki/Using-Ruffle#list-of-options):

> `salign` A string with a default value of "". This controls the position of the movie after scaling to fill the viewport and crops it as needed. The default alignment is centered. T, L, B, and R mean top, left, bottom, and right respectively. Can technically be any number of T's, L's, B's, and R's in any order, but the logical choices are "", "T", "L", "B", "R", "TL"/"LT", "TR"/"RT", "BL"/"LB", or "BR"/"RB". If you use some strange combination with both T's and B's or both L's and R's, keep in mind the precedence is L > R > "" and T > B > "".
> 
> `scale` A string with a default value of "showAll". The options are:
> * "showAll": Makes the entire SWF file visible in the specified area without distortion, while maintaining the original aspect ratio of the movie. Borders can appear on two sides of the movie.
> * "noborder": Scales the SWF file to fill the specified area, while maintaining the original aspect ratio of the file. The content may be cropped, but no distortion occurs.
> * "exactfit": Makes the entire SWF file visible in the specified area without trying to preserve the original aspect ratio. Distortion can occur.
> * "noscale": Prevents the SWF file from scaling to fit the area of the player. Cropping can occur.
> 
> `quality` A string with a default value of "high". Does nothing in Ruffle currently but can still be set and may do something in the future. The main options are "low", "medium", "high", and "best".

### CODE REVIEW

**Comment by midgleyc** [comment]
LGTM. Nice job finding a use in the wild!

One comment on style: in each of the Rust functions, you've got something like:
```
            if let Ok(qual) = StageQuality::from_str(quality) {
                stage.set_quality(context.gc_context, qual);
            }
```
i.e. you've used "qual" instead of "quality" to avoid shadowing the previous variable. As you don't use `quality` inside the `if let` block, I think you should just be able to use `quality` all the way down, which might look cleaner?

**Comment by danielhjacobs** [comment]
> i.e. you've used "qual" instead of "quality" to avoid shadowing the previous variable. As you don't use `quality` inside the `if let` block, I think you should just be able to use `quality` all the way down, which might look cleaner?

Done. I'm relatively new to Rust and wasn't sure that would work.

**Comment by danielhjacobs** [comment]
Actually: quick opinion question:

Do we think the names of the Rust functions' input parameters should match the input parameters of the stage functions (align, scale_mode, and quality) or the names of the config parameters (salign, scale, and quality)?

File: core/src/player.rs [hunk 1]
```diff
@@ -772,6 +775,33 @@ impl Player {
...
     }
 
+    pub fn set_salign(&mut self, salign: &str) {
+        self.mutate_with_update_context(|context| {
+            let stage = context.stage;
+            if let Ok(salign) = StageAlign::from_str(salign) {
+                stage.set_align(context, salign);
+            }
+        })
+    }
```
Comment (Herschel) [inline core/src/player.rs:785]: I say let's have the `Player` be the line between the Flash HTML attribute names and our internal attribute names  -- I like `set_stage_align` here.

**Review by Herschel — APPROVED** [review]
Thank you!

**Pull request merged**